snapcraft/internal/pluginhandler.py

```diff
@@ -756,7 +756,8 @@ def _generate_include_set(directory, includes):
     include_files = set()
     for include in includes:
         if '*' in include:
-            matches = glob.glob(os.path.join(directory, include))
+            pattern = os.path.join(directory, include)
+            matches = iglob(pattern, recursive=True)
```

Comment:
@kyrofa it should be. We go over the list only when making it a set on the line below